astropy/table/column.py

```diff
@@ -313,8 +313,11 @@ def __setstate__(self, state):
 
         state = state[:-1]
 
-        # Using super(type(self), self).__setstate__() gives an infinite
-        # recursion.  Manually call the right super class to actually set up
+        # Using super().__setstate__(state) gives
+        # TypeError 'int' object is not iterable
+        # astropy.table._column_mixins._ColumnGetitemShim.__setstate_cython__()
```

Comment:
Is that a remnant of an earlier attempt?

Reply:
I think the recursion was from an earlier attempt, before `ColumnGetitemShim` even existed. I thought I'd try what would happen if I changed it anyway, and there now is this new error. For another time...

Reply:
No I meant the `astropy.table._column_mixins._ColumnGetitemShim.__setstate_cython__()` line. That seems new but unrelated to the changes/comments, or am I wrong?

Reply:
Ah, the comment is not quite clear - that is where the `TypeError` is raised. I'll clarify.